### ISSUE TITLE
Vulkano needs a new active maintainer

### ISSUE THREAD

**Issue by cosmicchipsocket** [vulkano-rs/vulkano#1435]
From someone else on the project gitter:

> antonino @pac85_gitlab Sep 03 20:16
> I've noticed that commits suddently stoppeted on the 12th of June, is everything alright with the project?

> Eliah Lakhin @eliah_lakhin_twitter Sep 09 12:26
>> I've noticed that commits suddently stoppeted on the 12th of June, is everything alright with the project?
> 
> @pac85_gitlab Well, I'm not a maintainer of the project, just an active user and a minor contributor. @AustinJ235 has disappeared, unfortunately. I didn't see any activity in his own projects too for the past time. Hopefully, he will get back, but in worst case I think we can ask Tomaka to assign another one maintainer, or maybe even do our own fork. But in this moment there are no urgent PR pendings or bugs to be fixed urgently. The codebase is quite stable.

I am concerned about the wellbeing of this project, due to the lack of commits and the lack of response I've received to issues trying to use the library. Is there a way that the project can become active again? Does Tokama need to find a new maintainer, as Eliah mentioned?

**Comment by AustinJ235** [comment]
Eliah has been helping out some. He was appointment as a maintainer a couple weeks ago. I am currently working long hours at work so my freetime is a bit sparse currently. hopefully when this pandemic is over my hours will return to normal and I can out more on the project.

**Comment by Eliah-Lakhin** [comment]
@cosmicchipsocket, @MitchellHansen My apology that I wasn't able to perform active maintenance for the past time too.

I'm very busy with my personal project that is using Vulkano under the hood, and I'm also helping some other OSS Rust projects from time to time. But technically now I have an access to Vulkano repo to review and merge Pull Requests. And I'm interested in continuation of Vulkano in long term.

So, if you have something that requires operative review and delivery, please mention me in the PR or in Github Issues. Then I will get notification by email and will prioritize it. Or bring me a line to "eliah.lakhin [at] gmail.com" or by Telegram https://t.me/eliah_lakhin. Please don't hesitate to do so as often as you need to get my attention. I just can't keep an eye for the project constantly and can miss something important, but can switch to an issue relatively quickly whenever it needed.

> I am concerned about the wellbeing of this project

Well, my personal vision is that Vulkano is already in quite stable state to be used without the active development, I think it already passed an active development phase in the past and has comprehensive set of stabilized features at least to be used for Desktop applications. After all Vulkano is just a middleware between Vulkan and Rust, and the set of features of Vulkan itself that could be exposed to Rust is finite.

Of course we need to have some level of maintenance such as updating dependent crates and fixing edge-case bugs that could be discovered. Also Vulkan itself is slowly evolving too. So if someone needs to fix something or develop a new feature or maybe even do some refactoring, it is very welcome, and I will be glad to help and assist you as much as I can.

**Comment by Eliah-Lakhin** [comment]
@cosmicchipsocket I just reviewed your questions #1434 and #1433. After a brief look I'm afraid I can't give any suggestions. Maybe you can show us a link to the actual application code?

But also to mention, I didn't work much with Graphics Pipeline. In my project I use Vulkano for Compute Pipeline(this is a raymarching toolkit that utilizes GPU compute shaders only). So if there is something related or around of this stuff maybe I can give some comments. Anything aside of this is just out of my current expertise.

Basically when you face an issue you can try to dig inside Vulkano's code, try to figure out what's going on there and maybe provide a fix depending on the needs. That's what I do when I face some issues related to Compute Pipeline. And I will be happy to merge your contribution.

**Comment by LukasKalbertodt** [comment]
@Eliah-Lakhin 
> Well, my personal vision is that Vulkano is already in quite stable state to be used without the active development, I think it already passed an active development phase in the past and has comprehensive set of stabilized features at least to be used for Desktop applications.

This sounds somewhat different from what the README state under "development status":

> Vulkano is still in heavy development and doesn't yet meet its goals of being very robust. However the general structure of the library is most likely definitive, and all future breaking changes will likely be straight-forward to fix in user code.

Maybe the README needs updating? It would be very beneficial if the README could clearly communicate the current state of the library. Just as an example: someone might find the project, read the "development status" part of the README and then look at the contribution activity. Since [this graph](https://github.com/vulkano-rs/vulkano/graphs/contributors) doesn't look like "heavy development", people might conclude that the project has been given up on. 

On a similar note, while note really the responsibility of this crate's README, it would be beneficial to have a "should you use vulkano?" section somewhere. As far as I can tell, there are two "proper" Vulkan crates: vulkano and ash. Both are very different. Furthermore, there are other not-only-Vulkan crates, like gfx-rs and wgpu. It would be great to help people decide on a crate for their projects.

(Regardless, thanks for your work on this crate!)

**Comment by AustinJ235** [comment]
@LukasKalbertodt Probably wouldn't hurt to have the readme updated to this view. This is mainly how I see the project currently. It is very functional it could use some nice to haves (guided queue creation for example). Personally as a maintainer of the project I am active in regards to keeping it working and providing maintenance to keep it up to date. I never sought out to be the one to drive massive new reworks or features and have tried to stay along the idea the original creator Tomaka had intended. 

Vulkano, after all, is the graphics library used for my UI, Basalt. I don't foresee switching away from Vulkano any time soon and will maintain as part of my work on working on my adventures. From time to time I will have a peak of interest in advancing Vulkano to something better, but as for now recovering from my depression a few months ago my strive has been more towards my personal projects. The main project I am currently working is closed source hence I may appear inactive on my github profile, but I am actively developing it. This app uses Basalt which in turn uses Vulkano, so I do actively use this library and depend on it.

The project has strove away from being a single maintainer to a project that is maintained by its users. Nothing is to stop anyone from developing new features and submitting it for review or feedback. @Eliah-Lakhin and I are always reading pull requests and issues.

Hope this clears things up a little.

**Comment by Rua** [comment]
Hi! :D
